components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -402,6 +402,9 @@ void RewardsServiceImpl::InitPrefChangeRegistrar() {
       base::Bind(
           &RewardsServiceImpl::OnPreferenceChanged,
           base::Unretained(this)));
+  profile_pref_change_registrar_.Add(
+      ads::prefs::kEnabled, base::Bind(&RewardsServiceImpl::OnPreferenceChanged,
```

Comment:
Side note: eventually we need to take a look at the dependency cycles between the rewards service and the ads service and split the responsibilities a bit better.